src/parse/mod.rs

```diff
@@ -66,11 +66,21 @@ pub(crate) enum Stmt {
     KeyframesRuleSet(Box<KeyframesRuleSet>),
 }
 
+bitflags::bitflags! {
+    // todo: try to remove the flag IN_CONTROL_FLOW
+    /// Flags to indicate the context during parsing.
+    pub struct Flags: u8 {
+        const IN_MIXIN        = 1;
+        const IN_FUNCTION     = 1 << 1;
+        const IN_CONTROL_FLOW = 1 << 2;
+        const IN_KEYFRAMES    = 1 << 3;
+    }
```

Comment:
This could be an enum if it is mutual exclusive but I don't think that is the case.

Reply:
`in_mixin`, `in_function`, and `in_control_flow` are mutually exlcusive, but that shouldn't be too much of an issue.

I am a bit wary of bringing in a new dependency, though the usage of `bitflags` here should be easy to refactor out (that is, manually implement our own `ContextFlags` struct). For right now this should be fine, but ideally we would remove the dependency before `0.10.0`

Reply:
I've removed the `bitflags` dependency here, https://github.com/connorskees/grass/commit/92aa149f8a8bdc1f1710d4859994c769675473de